extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/batch/serializer/JoinBatchInformationTypeAdapter.java

```diff
@@ -60,6 +61,10 @@ public JsonElement serialize(JoinBatchInformation batch, Type typeOfSrc, JsonSer
         result.add("span", new JsonPrimitive(span.getStart().getsRow() + "\u0000" + span.getEnd().getsRow()));
         result.add("startInc", new JsonPrimitive(span.isStartInclusive()));
         result.add("endInc", new JsonPrimitive(span.isEndInclusive()));
+        if(batch.getAggregationStateMeta().isPresent()) {
+            CommonNodeMetadataImpl stateMeta = batch.getAggregationStateMeta().get();
+            result.add("aggStateMeta", new JsonPrimitive(stateMeta.getNodeId() + "\u0000" + stateMeta.getVariableOrder().toString()));
```

Comment:
Maybe move the null character somewhere common.

Reply:
Done